### PR TITLE
cw20-merkle-airdrop: change hashing to sha256

### PR DESCRIPTION
**Pull request by orkunkl**

### CODE REVIEW

File: contracts/cw20-merkle-airdrop/helpers/README.md [hunk 1]
```diff
@@ -26,12 +26,12 @@ Binary will be placed to path.
...
 **Generate Root:**
 ```shell
-merkle-airdrop-cli generateRoot --file ../testdata/airdrop_stage_2.json
+merkle-airdrop-cli generateRoot --file ../testdata/airdrop_stage_2_list.json
 ```
 
 **Generate proof:**
 ```shell
-merkle-airdrop-cli generateProofs --file ../testdata/airdrop_stage_2.json \
+merkle-airdrop-cli generateProofs --file ../testdata/airdrop_stage_2_list.json \
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/helpers/README.md:34]: Ah, did you use the output of these as input to rust tests?

It does look like you have a place to generate test vectors.

File: contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts [hunk 2]
```diff
@@ -18,23 +18,22 @@ class Airdrop {
...
   ): boolean {
-    let hashBuf = keccak256(account.address + account.amount);
+    let hashBuf = Buffer.from(sha256(account.address + account.amount).toString())
 
     proof.forEach((proofElem) => {
       const proofBuf = Buffer.from(proofElem, 'hex');
-
       if (hashBuf < proofBuf) {
-        hashBuf = keccak256(Buffer.concat([hashBuf, proofBuf]));
+        hashBuf = Buffer.from(sha256(Buffer.concat([hashBuf, proofBuf]).toString()));
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts:34]: why type does sha256 return? Why `Buffer.from(sha256(a).toString())`?
Seems like there may be an easier way to transform it.

(Also if using Buffer.from, I think best to specify the input type. I think the default is `hex`, but let's be specific)

File: contracts/cw20-merkle-airdrop/src/contract.rs [hunk 3]
```diff
@@ -16,6 +15,7 @@ use crate::msg::{
     MerkleRootResponse, MigrateMsg, QueryMsg,
 };
 use crate::state::{Config, CLAIM, CONFIG, LATEST_STAGE, MERKLE_ROOT};
+use sha2::Digest;
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/src/contract.rs:18]: I'd move this up, same line where sha3 was

File: contracts/cw20-merkle-airdrop/src/contract.rs [hunk 4]
```diff
@@ -359,9 +360,33 @@ mod tests {
         );
     }
 
+    const TEST_DATA_1: &str = "./testdata/airdrop_stage_1_test_data.json";
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/src/contract.rs:363]: I like it!

Comment (ethanfrey) [reply contracts/cw20-merkle-airdrop/src/contract.rs:363]: Looking above, I don't see the command that generates these test data.
Can you please add this to the README of the ts side (or a comment here)?

File: contracts/cw20-merkle-airdrop/src/contract.rs [hunk 5]
```diff
@@ -359,9 +360,33 @@ mod tests {
...
+
+    #[derive(Deserialize, Debug)]
+    struct Encoded {
+        account: String,
+        amount: Uint128,
+        root: String,
+        proofs: Vec<String>,
+    }
+
+    fn read_test_data(path: &str) -> Encoded {
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/src/contract.rs:374]: There is a nice trick you can use...

Replace:

```rust
const TEST_DATA_1: &str = "./testdata/airdrop_stage_1_test_data.json";
let test_data = read_test_data(TEST_DATA_1);
```

with

```rust
const TEST_DATA_1: &[u8] = include_bytes!("./testdata/airdrop_stage_1_test_data.json");
let test_data: Encoded = from_slice(TEST_DATA_1).unwrap();
```

Then you can remove `read_test_data`.

File: contracts/cw20-merkle-airdrop/src/contract.rs [hunk 6]
```diff
@@ -375,30 +400,25 @@ mod tests {
...
         let msg = ExecuteMsg::Claim {
-            amount: Uint128::from(10220u128),
+            amount: test_data.amount,
             stage: 1u8,
-            proof: vec![
-                "561b09601236f43b0065d1ea601df754989911ed6da16ba8afed62c458f86352".to_string(),
-                "280777995d054081cbf208bccb70f8d736c1766b81d90a1fd21cd97d2d83a5cc".to_string(),
-                "3946ea1758a5a2bf55bae1186168ad35aa0329805bc8bff1ca3d51345faec04a".to_string(),
-            ],
+            proof: test_data.proofs,
```
Comment (ethanfrey) [inline contracts/cw20-merkle-airdrop/src/contract.rs:410]: Nice to use this from those files.

**Review by ethanfrey — APPROVED** [review]
Good stuff. Some nit-pick comments to improve coding style, but it looks good.

Glad to see the test vector. Can you just document the command you use in the TS code to generate that test data?

File: contracts/cw20-merkle-airdrop/helpers/README.md (hunk 1 above)
Comment (orkunkl) [reply contracts/cw20-merkle-airdrop/helpers/README.md:34]: Yep exactly

File: contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts (hunk 2 above)
Comment (orkunkl) [reply contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts:34]: sha256 returns `CryptoJS/WordArray` which is hard to work with. Only `toString` method available.

**Comment by orkunkl** [comment]
Merging now, helper.ts works contract works. I will leave the rest to CosmJS DevRel person!

File: contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts (hunk 2 above)
Comment (ethanfrey) [reply contracts/cw20-merkle-airdrop/helpers/src/airdrop.ts:34]: Well, best you could do with what you got.

**Pull request merged**